node/src/reactor/main_reactor/catch_up.rs

```diff
@@ -333,37 +332,35 @@ impl MainReactor {
         self.block_accumulator.reset_last_progress();
```

Comment:
Seems like this should only be done if we do actually leap.  It would be good to move it inside of `request_leap_if_not_redundant`.